plugins/wacom-raw/fu-wacom-device.c

```diff
@@ -132,16 +132,20 @@ fu_wacom_device_attach(FuDevice *device, FuProgress *progress, GError **error)
 				 .cmd = FU_WACOM_RAW_BL_CMD_ATTACH,
 				 .echo = FU_WACOM_RAW_ECHO_DEFAULT,
 				 0x00};
-	if (!fu_device_has_flag(device, FWUPD_DEVICE_FLAG_IS_BOOTLOADER)) {
-		g_debug("already in runtime mode, skipping");
-		return TRUE;
-	}
-	if (!fu_wacom_device_set_feature(self, (const guint8 *)&req, sizeof(req), error)) {
-		g_prefix_error(error, "failed to switch to runtime mode: ");
-		return FALSE;
+	gboolean emr_device = fu_device_has_guid(device, "WacomEMR");
+
+	/* only EMR requires way back to runtime mode */
+	if (emr_device) {
```

Comment:
this should move to fu-wacom-emr-device.c -- if the aes and emr devices have different attach behaviours this should be moved into the subclassed functions.

Reply:
Yes, now they should have different process to get back to their boot loader. I'll separate out the functions for each EMR or AES; I'll create wacom_aes_attach and wacom_emr_attach subclasses.